### PR TITLE
#MES-4510 - feat(CatAMod2-TestReport): implement UI changes to display complete C…

### PR DESCRIPTION
**Pull request by AndyPerryKainos**
…atAMod2 test report

fixes story #MES-4510

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [x] One review from each scrum team
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/assets/i18n/cy.json [hunk 1]
```diff
@@ -285,6 +303,7 @@
       "forwardParkControl": "Parcio Ymlaen - dan reolaeth",
       "forwardParkObservation": "Parcio Ymlaen - gwyliadwriaeth",
       "vehicleChecks": "Gwirio’r cerbyd",
+      "safetyAndBalance": "Safety and balance questions",
```
Comment (turonekt) [inline src/assets/i18n/cy.json:306]: Welsh translation is missing

File: src/modules/tests/test-data/test-data.constants.ts [hunk 2]
```diff
@@ -1,12 +1,17 @@
 export enum Competencies {
   controlsAccelerator = 'controlsAccelerator',
+  controlsBalanceSlowControl = 'controlsBalanceSlowControl',
   controlsClutch = 'controlsClutch',
   controlsGears = 'controlsGears',
   controlsFootbrake = 'controlsFootbrake',
+  controlsFrontBrake = 'controlsFrontBrake',
```
Comment (szabikr) [inline src/modules/tests/test-data/test-data.constants.ts]: controlsFrontbreak

**Review by faggc — APPROVED** [review]
Looks fine to me. Just waiting on the missing Welsh translation.

File: src/providers/fault-count/cat-a-mod2/fault-count.cat-a-mod2.ts [hunk 3]
```diff
@@ -1,10 +1,51 @@
...
-import { SafetyAndBalanceQuestions, QuestionResult } from '@dvsa/mes-test-schema/categories/AM2';
+import { TestData, SafetyAndBalanceQuestions, QuestionResult } from '@dvsa/mes-test-schema/categories/AM2';
 
 export class FaultCountAM2Helper {
 
+  public static getRidingFaultSumCountCatAM2 = (data: TestData): number => {
+
+    // The way how we store serious faults differs for certain competencies
+    // Because of this we need to pay extra attention on summing up all of them
+    // const { drivingFaults, seriousFaults, dangerousFaults, safetyAndBalanceQuestions } = data;
```
Comment (szabikr) [inline src/providers/fault-count/cat-a-mod2/fault-count.cat-a-mod2.ts:12]: Just delete this code.

**Review by szabikr — APPROVED** [review]
Just remove all commented out code

File: src/modules/tests/test-data/cat-a-mod2/_tests_/test-data.cat-a-mod2.selector.spec.ts [hunk 4]
```diff
@@ -1,253 +1,254 @@
...
-            },
-          ],
-        } as SafetyAndBalanceQuestions;
-
-        expect(hasVehicleChecksBeenCompleted(state)).toEqual(false);
-      });
-    });
-  });
-});
+// import { TestData, SafetyAndBalanceQuestions } from '@dvsa/mes-test-schema/categories/AM2';
```
Comment (TomBillingtonUK) [inline src/modules/tests/test-data/cat-a-mod2/_tests_/test-data.cat-a-mod2.selector.spec.ts:1]: Are there plans to fix these?

File: src/pages/debrief/cat-a-mod2/components/safety-and-balance-card/__tests__/vehicle-checks-card.cat-a-mod2.spec.ts [hunk 5]
```diff
@@ -1,15 +1,15 @@
 import { ComponentFixture, async, TestBed } from '@angular/core/testing';
-import { VehicleChecksCardCatAMod2Component } from '../vehicle-checks-card.cat-a-mod2';
+import { SafetyAndBalanceCardCatAMod2Component } from '../safety-and-balance-card.cat-a-mod2';
 import { IonicModule, Config } from 'ionic-angular';
 import { StoreModule, Store } from '@ngrx/store';
 import { testsReducer } from '../../../../../../modules/tests/tests.reducer';
 import { StoreModel } from '../../../../../../shared/models/store.model';
 import { StartTest } from '../../../../../../modules/tests/tests.actions';
 // TODO - PREP-AMOD2 - Implement category specific actions
```
Comment (TomBillingtonUK) [inline src/pages/debrief/cat-a-mod2/components/safety-and-balance-card/__tests__/vehicle-checks-card.cat-a-mod2.spec.ts:8]: TODO which can go

Comment (AndyPerryKainos) [reply src/pages/debrief/cat-a-mod2/components/safety-and-balance-card/__tests__/vehicle-checks-card.cat-a-mod2.spec.ts:8]: Entire file has wrong name. Fixing.

File: src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      select(getCurrentTest),
+    );
+
+    this.componentState = {
+      safetyAndBalanceDrivingFaultCount$: currentTest$.pipe(
+        select(getTestData),
+        select(getSafetyAndBalanceQuestions),
+        map((safetyAndBalance) => {
+          return this.faultCountProvider
+            .getSafetyAndBalanceFaultCount(TestCategory.EUA2M2, safetyAndBalance)
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts]: I don't think we should be sending in a hard coded category it should grab the category from the store and pass it in.

File: src/modules/tests/test-data/cat-a-mod2/_tests_/test-data.cat-a-mod2.selector.spec.ts (hunk 4 above)
Comment (AndyPerryKainos) [reply src/modules/tests/test-data/cat-a-mod2/_tests_/test-data.cat-a-mod2.selector.spec.ts:1]: Yes, MES-4515.

File: src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts (hunk 6 above)
Comment (AndyPerryKainos) [reply src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts]: Yup. On it.

**Review by TomBillingtonUK — APPROVED** [review]
Happy to approve, a couple of comments which can be addressed.

File: src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts (hunk 6 above)
Comment (AndyPerryKainos) [reply src/pages/test-report/cat-a-mod2/components/safety-and-balance/safety-and-balance.ts]: Sorted. Rebasing and pushing now.

**Pull request merged**